Shell/Program.cs

```diff
@@ -26,7 +26,7 @@
 builder.AddTransient<IShellCommand, ShellCommand>();
 builder.AddTransient<ICommandFactory, CommandFactory>();
 
-builder.AddTransient<ILexer, Lexer.Lexer>();
+builder.AddSingleton<ILexer, Lexer.Lexer>();
```

Comment:
This AddSingleton means that everytime you ask via the DI for a container you will receive the same object ( its not newed if it already exists )...if register the Lexer as a singleton the Lexer class self doesn't need to be a singleton class... It should be a normal class because the DI container is handling the singleton stuff. So the singleton stuff needs to be removed from the Lexer class.